src/bisection.ts

```diff
@@ -130,14 +146,14 @@ export class ChallengeManager {
     this.consensusStep = newConsensusStep;
     this.disputedStep = newDisputedStep;
     this.stateHashes = [consensusWitness.witness, ...hashes];
-
+    this.root = generateRoot(this.stateHashes);
     this.caller = caller;
   }
 
   detectFraud(
-    {witness: consensusWitness}: Witness,
+    {witness: consensusWitness}: WitnessProof,
```

Comment:
These witnesses should be validated.

Reply:
Haha oops. I made this change locally in 5ae89bf but forgot to push 🤦